traffic_ops/traffic_ops_golang/capabilities/capabilities.go

```diff
@@ -0,0 +1,305 @@
+package capabilities
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import "database/sql"
+import "encoding/json"
+import "errors"
+import "fmt"
+import "net/http"
+
+import "github.com/apache/trafficcontrol/lib/go-tc"
+import "github.com/apache/trafficcontrol/traffic_ops/traffic_ops_golang/api"
+
+const readQuery = `
+SELECT description,
+       last_updated,
+       name
+FROM capability
+`
+
+const createQuery = `
+INSERT INTO capability (name, description)
+VALUES ($1, $2)
+RETURNING description, last_updated, name
+`
+
+const replaceQuery = `
+UPDATE capability
+SET name=$1, description=$2
+WHERE name=$3
+RETURNING description, last_updated, name
+`
+
+const deleteQuery = `
+DELETE FROM capability
+WHERE name=$1
+RETURNING description, last_updated, name
+`
+
+func Read(w http.ResponseWriter, r *http.Request) {
+	inf, sysErr, userErr, errCode := api.NewInfo(r, nil, nil)
+	tx := inf.Tx.Tx
+	if userErr != nil || sysErr != nil {
+		api.HandleErr(w, r, tx, errCode, userErr, sysErr)
+		return
+	}
+	defer inf.Close()
+
+
+	var rows *sql.Rows
+	var err error
+	if name, ok := inf.Params["name"]; ok {
+		rows, err = tx.Query(readQuery + "WHERE name=$1", name)
```

Comment:
could we instead use https://github.com/apache/trafficcontrol/blob/master/traffic_ops/traffic_ops_golang/dbhelpers/db_helpers.go#L45?